src/app/dashboard/dashboard.component.ts

```diff
@@ -1,8 +1,48 @@
-import { Component, OnInit } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
+import {ApiService} from '../Services/api.service';
+import {HttpErrorResponse} from '@angular/common/http';
+import {BalanceDto} from '../balanceDto';
 
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: ['./dashboard.component.css'],
 })
-export class DashboardComponent {}
+export class DashboardComponent implements OnInit {
+  constructor(private apiService: ApiService) {}
+
+  responseStatusClockIn: number = null;
+  responseStatusClockOut: number = null;
```

Comment:
Jak już kiedyś wcześniej wspomniałam, null jest domyślną wartością podczas definiowania jeśli nie przypisujemy żadnej wartości dlatego wystarczy

```suggestion
  responseStatusClockIn: number;
  responseStatusClockOut: number;
```

Reply:
A czy domyślną wartością nie jest undefined? Bez przypisanego nulla warunek nie mi działał. Można też warunek zmienić na undefined.

Reply:
Dlatego własnie zamiast w warunku sprawdzać czy określone pole jest nullem (tak jak w dashboard.component.html), możemy użyć sprawdzenia poprzez nazwaPola/!nazwaPola, co sprawdzi czy wartość w polu jest lub nie jest "falsy" (czyli NaN, null, undefined, empty string, 0n, -0, 0, false).

Reply:
Ok, już rozumiem dzięki 😉

Reply:
Nie jest to oczywiście coś obowiązkowego, jeśli nie chcecie tego zmieniać może zostać. Jak wcześniej wspomniałam aktualne komentarze to sugestie dotyczące dobrych praktyk.

Reply:
Skasowałem nadmiarowe inicjalizacje nullami. Niestety warunek z samym !responseStatusClockIn coś dalej mi nie działał, może jeszcze coś źle zrobiłem 😅. Na razie zostawiłem ze sprawdzaniem, czy nie jest undefined.